### DISCUSSION TITLE
Updating pivot relation data in table action

### DISCUSSION THREAD

**Discussion by Jam0r85** [filamentphp/filament#3189] [Help]
I can't seem to wrap my head about this.

I have a BelongsToMany relationship of Clients to a Landlord and i'm using filament table to display a list of clients attached. In addition to this I have some pivot fields which i'd like to update via an action on this table.

I've created the action, added one form element and updating the pivot field works however I cannot seem to set the state of the form from the pivot field. Any ideas? I can manually set it false or true using the $component->state method but if I try and access the $record there is no pivot data.

Again I don't want to add or remove clients, it's more about updating the pivot fields per client to the landlord.

```
    protected function getTableActions(): array
    {
        return [
            Action::make('settings')
                ->action(function (Client $record, array $data) {
                    $this->landlord->clients()->updateExistingPivot($record->id, [
                        'statement_notifications' => $data['statement_notifications']
                    ]);
                })
                ->form([
                    Checkbox::make('statement_notifications')
                        ->afterStateHydrated(function (Client $record, $component) {
                            return $component->state(false);
                        })
                ])->modalWidth('lg'),

            Action::make('view')
                ->url(fn (Client $record): string => route('clients.dashboard', $record)),
        ];
    }
```

This is how it looks

![Screenshot 2022-07-20 at 20 23 34](https://user-images.githubusercontent.com/18726355/180065153-4971fbf2-164b-4310-ae1a-80adfb3f8033.jpg)

![Screenshot 2022-07-20 at 20 23 43](https://user-images.githubusercontent.com/18726355/180065164-a4f9d8a3-b222-4dbf-bbaa-a40a13c73789.jpg)

**Comment by danharrin** [comment]
`->mountUsing(fn ($record, $form) => $form->fill(['statement_notifications' => $record->pivot-> statement_notifications]))` maybe?
